app/Export/GetImages.hs

```diff
@@ -15,43 +15,44 @@ import qualified Data.Text as T
 import System.Random
 import Svg.Generator
 import Export.ImageConversion
-import Happstack.Server (Request, rqCookies, cookieValue)
-import Data.List.Utils (replace)
 import Database.CourseQueries (getMeetingTime)
 import Database.Tables as Tables
 import Database.Persist.Sqlite (runSqlite)
 import Config (databasePath)
 import Data.Fixed (mod')
+import Data.Aeson (decode)
+import Data.ByteString.Char8 as BC (pack)
+import Data.ByteString.Lazy (fromStrict)
+import Data.Maybe (fromMaybe, fromJust)
+
 
 -- | If there is an active graph available, an image of that graph is created,
 -- otherwise the Computer Science graph is created as a default.
 -- Either way, the resulting graph's .svg and .png names are returned.
-getActiveGraphImage :: Request -> IO (String, String)
-getActiveGraphImage req = do
-    let cookies = M.fromList $ rqCookies req
-        graphName =
-            T.pack $
-            replace "-" " " $
-                maybe "Computer-Science" cookieValue (M.lookup "active-graph" cookies)
-    getGraphImage graphName (M.mapKeys T.pack $ M.map (T.pack . cookieValue) cookies)
+getActiveGraphImage :: String -> IO (String, String)
+getActiveGraphImage jsonLocalStorage = do
+    let localStorage = fromJust $ decode $ fromStrict $ BC.pack jsonLocalStorage :: M.Map T.Text T.Text
+        graphName = T.replace "-" " " $ fromMaybe "Computer-Science" $ M.lookup "active-graph" localStorage
```

Comment:
When we used to use cookies, we had trouble getting spaces in the cookies to be parsed. I'm wondering if this is no longer true. Can you please try to store the graph names with spaces instead of hyphens (e.g., "Computer Science") and see if the loading works properly?